tests/lib/thrift/ThriftTest/src/client.cpp

```diff
@@ -0,0 +1,509 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#include <array>
+#include <cmath>
+#include <limits>
+#include <locale>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+
+#include <pthread.h>
+#include <signal.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/server/TSimpleServer.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TSocket.h>
+
+#include "ThriftTest.h"
+
+#include "TestUtils.hpp"
+
+using namespace std;
+
+using namespace apache::thrift;
+using namespace apache::thrift::async;
+using namespace apache::thrift::concurrency;
+using namespace apache::thrift::protocol;
+using namespace apache::thrift::transport;
+using namespace apache::thrift::server;
+using namespace thrift::test;
+
+#define HELLO_SERVER_STACK_SIZE 4096
+static K_THREAD_STACK_DEFINE(ThriftTest_server_stack, HELLO_SERVER_STACK_SIZE);
+
+struct ctx {
+  shared_ptr<ThriftTestClient> client;
+  shared_ptr<TServer> server;
+  pthread_t server_thread;
+};
+static ctx context;
+
+shared_ptr<TServer> setup_server(const std::string& my_addr, const int port);
+
+static void init_Xtruct(Xtruct& s);
+
+static void binary_fill(std::string& str, string::size_type siz);
+
+static uint64_t now();
+
+static void void_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  try {
+    testClient.testVoid();
+  } catch (TTransportException&) {
+    // Stop here if transport got broken
+    throw;
+  } catch (exception& ex) {
+    zassert_exception("%s", ex.what());
+  }
+}
+
+static void string_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  string s;
+  testClient.testString(s, "Test");
+  zassert_equal(s, "Test", "Unexpected string %s, expected \"Test\"", s.c_str());
+}
+
+static void bool_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  BASETYPE_IDENTITY_TEST(testBool, testClient, true);
+  BASETYPE_IDENTITY_TEST(testBool, testClient, false);
```

Comment:
```suggestion
  zassert_equal(false, context.client->testBool(false), "");
  zassert_equal(true, context.client->testBool(true), "");
```

Reply:
There are lots of similar statements below. Let me convert them.